common/metric_types.py

```
@@ -288,6 +290,38 @@ async def _send_request(
+class EVMBlockNumberLatencyMetric(HttpCallLatencyMetricBase):
+    """Shared eth_blockNumber metric used by all EVM chains.
+
+    Captures the returned block number for cross-provider lag computation.
+    """
+
+    @property
+    def method(self) -> str:
+        return "eth_blockNumber"
+
+    @staticmethod
+    def get_params_from_state(state_data: dict[str, Any]) -> list[Any]:
```

Ruff (0.15.9)

[warning] Unused static method argument: `state_data`

(ARG004)